registry/storage/driver/gcs/gcs.go

```diff
@@ -689,42 +676,52 @@ func (d *driver) List(ctx context.Context, path string) ([]string, error) {
 // Move moves an object stored at sourcePath to destPath, removing the
 // original object.
 func (d *driver) Move(ctx context.Context, sourcePath string, destPath string) error {
-	_, err := storageCopyObject(ctx, d.bucket, d.pathToKey(sourcePath), d.bucket, d.pathToKey(destPath), d.gcs)
+	srcKey, dstKey := d.pathToKey(sourcePath), d.pathToKey(destPath)
+	src := d.bucket.Object(srcKey)
+	_, err := d.bucket.Object(dstKey).CopierFrom(src).Run(ctx)
 	if err != nil {
-		if status, ok := err.(*googleapi.Error); ok {
+		var status *googleapi.Error
+		if errors.As(err, &status) {
 			if status.Code == http.StatusNotFound {
-				return storagedriver.PathNotFoundError{Path: sourcePath}
+				return storagedriver.PathNotFoundError{Path: srcKey}
 			}
 		}
-		return err
+		return fmt.Errorf("move %q to %q: %w", srcKey, dstKey, err)
```

Comment:
Is it safe to wrap the error here? I see we're also returning typed errors; is there any calling code that could check for the original error (but not expecting it to be wrapped?)

I did a quick check, and couldn't find at a glance, but seen quite some straight compares for error-types, which is why I commented 😅

Reply:
wrapping isn't needed here - I can't remember if this is a typo by me 🤔 
changed here: https://github.com/distribution/distribution/pull/4120/commits/7ee9dcac685cc0bbcf332f5387c5dec4f75e9058